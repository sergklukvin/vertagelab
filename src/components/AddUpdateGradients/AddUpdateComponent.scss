.input-control {
  display: grid;
  grid-row-gap: 5px;
  grid-auto-rows: max-content;
}
  
.error {
  color: red;
  font-size: 12px;
}