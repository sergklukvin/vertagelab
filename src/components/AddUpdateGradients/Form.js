import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  createColor,
  getAllColors,
  updateColor,
} from '../../services/http.service';
import { useForm } from 'react-hook-form';
import './Form.scss';
import { Redirect } from 'react-router-dom';
import { AlertBar } from '../Alert/AlertBar';

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1),
      width: '80%',
    },
  },
  mainBlock: {
    marginTop: 30,
    marginBottom: 30,
  },
  btn: {
    color: 'white',
  },
  btnSubmit: {
    cursor: 'pointer',
  },
  btnDisable: {
    backgroundColor: 'grey',
  },
}));

export default function MultilineTextFields(props) {
  let curId = '';
  const classes = useStyles();
  const listGradients = props.listGradients;
  const [redirect, setRedirect] = useState(false);
  const [alertMeta, setAlertMeta] = useState({ severity: '', message: '' });
  const [isListManipulated, setListManipulated] = useState(false);
  const [isAlertOpen, setAlertOpen] = useState(false);

  const { register, handleSubmit, errors } = useForm();

  const onSubmit = (data) => {
    if (props.listGradients.id !== '') {
      updateColor(props.listGradients.id, {
        id: props.listGradients.id,
        colorOne: data.colorOne.toLowerCase(),
        colorTwo: data.colorTwo.toLowerCase(),
      })
        .then(() => showAlert('success', 'Update success'))
        .catch(() => showAlert('error', 'Update failed'))
        .then(
          setTimeout(() => {
            setRedirect(true);
          }, 1000)
        );
    } else {
      getAllColors()
        .then((res) => {
          if (res.data.length === 0) {
            curId = 1;
          } else {
            curId = res.data[res.data.length - 1].id + 1;
          }
        })
        .then(
          createColor({
            id: curId,
            colorOne: data.colorOne.toLowerCase(),
            colorTwo: data.colorTwo.toLowerCase(),
          })
        )
        .then(() => showAlert('success', 'Create success'))
        .catch(() => showAlert('error', 'Create error'))
        .then(
          setTimeout(() => {
            setRedirect(true);
          }, 1000)
        );
    }
  };

  const showAlert = (severity, message) => {
    setListManipulated(true);
    setAlertMeta({ severity, message });
    setAlertOpen(true);

    setTimeout(() => {
      handleAlertClose();
    }, 2000);
  };

  const handleAlertClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setAlertOpen(false);
  };

  if (redirect) {
    return (
      <div>
        <Redirect to='/' />
      </div>
    );
  }

  return (
    <>
      <AlertBar
        open={isAlertOpen}
        onClose={handleAlertClose}
        alertMeta={alertMeta}
      />

      <form onSubmit={handleSubmit(onSubmit)} className={classes.root}>
        <input
          name='colorOne'
          ref={register({ required: true, pattern: /#[a-fA-F0-9]{3,6}\b/gi })}
          placeholder='Color 1'
          defaultValue={listGradients.colorOne}
        />
        {errors.colorOne && <p>Field 'Color 1' is required</p>}

        <input
          name='colorTwo'
          ref={register({ required: true, pattern: /#[a-fA-F0-9]{3,6}\b/gi })}
          placeholder='Color 2'
          defaultValue={listGradients.colorTwo}
        />
        {errors.colorTwo && <p>Field 'Color 2' is required</p>}

        <input type='submit' value='SUBMIT' className={classes.btnSubmit} />
      </form>
    </>
  );
}
