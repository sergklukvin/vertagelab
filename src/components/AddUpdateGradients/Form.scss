select,
input {
  display: block;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  border: 1px solid grey;
  padding: 10px 15px;
  margin-bottom: 15px;
  font-size: 20px;
}

.label,
section > label {
  line-height: 2;
  text-align: left;
  display: block;
  margin-bottom: 13px;
  margin-top: 20px;
  color: white;
  font-size: 14px;
  font-weight: 200;
}

input[type="submit"],
.button {
  background: black;
  color: white;
  border: none;
  margin-top: 20px;
  padding: 20px;
  font-size: 16px;
  font-weight: 100;
  display: block;
  appearance: none;
  border-radius: 4px;
  width: 100%;
  font-weight: lighter;
}

input[type="button"],
.button {
  background: grey;
  color: white;
  border: none;
  margin-top: 20px;
  padding: 20px;
  font-size: 16px;
  font-weight: 100;
  display: block;
  appearance: none;
  border-radius: 4px;
  width: 100%;
  font-weight: lighter;
}

.container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
}

.counter {
  font-weight: 400;
  color: black;
  padding: 10px 15px;
  border-radius: 4px;
  position: fixed;
  top: 20px;
  right: 30px;
  z-index: 9999999;
}

button[type="button"] {
  padding: 5px;
  color: white;
  letter-spacing: 0px;
  padding: 10px;
  letter-spacing: 2px;
}

button[type="button"]:hover {
  background: black;
  color: white;
}

p {
  color: rgb(168, 31, 31);
  margin-bottom: 25px;
}
