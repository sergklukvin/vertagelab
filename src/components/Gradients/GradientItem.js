import React from 'react';
import './GradientItem.scss'


function GradientItem({data}) {

    return (
        <div className="gradientBox" style = {{background: `linear-gradient(to right, ${data[0]}, ${data[1]})`}}>
            <span>{data[0]}</span>
            <span>{data[1]}</span>
        </div>
    )
}

export default GradientItem
