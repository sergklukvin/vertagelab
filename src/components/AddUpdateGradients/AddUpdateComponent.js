import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import { getColor } from '../../services/http.service';
import Form from './Form';

const useStyles = makeStyles({
  mainBlock: {
    marginTop: 30,
    marginBottom: 30,
  },
  btn: {
    textDecoration: 'none',
    color: 'black',
    backgroundColor: 'black',
  },
});

function AddUpdateComponent(props) {

  const classes = useStyles();
  const [info, setInfo] = useState(null);

  useEffect(() => {
    let cleanupFunction = false;

    if (
      props.location.checkAdd === true ||
      props.location.gradientId === undefined
    ) {
      setInfo({id:'', colorOne: '', colorTwo: ''});
    } else {
      getColor(props.location.gradientId).then((res) => {
        if (!cleanupFunction) setInfo(res.data);
      });
    }

    return () => (cleanupFunction = true);
  }, []);

  if (info === null) {
    return <div>Loading...</div>;
  }

  return (
    <div>
      <div className={classes.mainBlock}>
        <Link to='/' className={classes.btn}>
          <Button className={classes.btn}>GO BACK</Button>
        </Link>
      </div>
      <div className={classes.mainBlock}>
        <Form listGradients={info} />
      </div>
    </div>
  );
}

export default AddUpdateComponent;
