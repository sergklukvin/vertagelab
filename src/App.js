import React from 'react';
import HeaderCard from './components/Header/HeaderCard';
import ListBooks from './components/Gradients/ListGradients';
import Container from '@material-ui/core/Container';
import AddUpdateComponent from './components/AddUpdateGradients/AddUpdateComponent';
import { BrowserRouter as Router, Switch, Route} from 'react-router-dom';

function App() {
  return (
    <Container maxWidth='md'>
      <HeaderCard />
      <Router>
        <Switch>
          <Route exact path='/' component={ListBooks} />
          <Route path='/add' component={AddUpdateComponent} />
        </Switch>
      </Router>
    </Container>
  );
}

export default App;
