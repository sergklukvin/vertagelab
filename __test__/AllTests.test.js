import React from 'react';
import App from '../src/App';
import GradientItem from '../src/components/Gradients/GradientItem';
import AddUpdateComponent from '../src/components/Gradients/ListGradients';
import Form from '../src/components/AddUpdateGradients/Form';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Redirect } from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('Vertagelab tests', () => {
  it('render correctly App component', () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
  });

  it('render redirect FormComponent', () => {
    const wrapper = shallow(
      <Form listGradients={{ id: '', colorOne: '', colorTwo: '' }}></Form>
    );
    expect(wrapper.find(Redirect)).toBeTruthy();
  });

  it('render correctly AddUpdateComponent component', () => {
    const AddUpdate = renderer
      .create(
        <AddUpdateComponent
          props={{ location: { pathname: '/add', checkAdd: true } }}
        />
      )
      .toJSON();
    expect(AddUpdate).toMatchSnapshot();
  });

  it('GradientItem should contain gradientBox-class', () => {
    const component = shallow(<GradientItem data={[1, 2]} />);
    const wrapper = component.find('.gradientBox');
    expect(wrapper.length).toBe(1);
  });

  it('Form should contain tag <form>', () => {
    const component = shallow(
      <Form listGradients={{ id: '', colorOne: '', colorTwo: '' }} />
    );
    const wrapper = component.find('form');
    expect(wrapper.length).toBe(1);
  });

  it('Form should contain tag <input>', () => {
    const component = shallow(
      <Form listGradients={{ id: '', colorOne: '', colorTwo: '' }} />
    );
    const wrapper = component.find('input');
    expect(wrapper.length).toBe(3);
  });
});
