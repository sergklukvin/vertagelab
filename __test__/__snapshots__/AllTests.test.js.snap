// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertagelab tests render correctly AddUpdateComponent component 1`] = `
<div>
  Loading...
</div>
`;

exports[`Vertagelab tests render correctly App component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
      >
        GRADIENTS
      </p>
    </div>
  </div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`Vertagelab tests render correctly Form component 1`] = `
<form
  className="makeStyles-root-11"
  onSubmit={[Function]}
>
  <input
    defaultValue=""
    name="colorOne"
    placeholder="Color 1"
  />
  <input
    defaultValue=""
    name="colorTwo"
    placeholder="Color 2"
  />
  <input
    className="makeStyles-btnSubmit-14"
    type="submit"
    value="SUBMIT"
  />
</form>
`;

exports[`Vertagelab tests the Button text test 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    GO BACK
  </span>
</button>
`;
